packages/cli/src/util/dev/server.ts

```diff
@@ -1566,9 +1567,31 @@ export default class DevServer {
 
             // Retain orginal pathname, but override query parameters from the rewrite
             const beforeRewriteUrl = req.url || '/';
-            const rewriteUrlParsed = url.parse(beforeRewriteUrl);
-            rewriteUrlParsed.search = url.parse(rewritePath).search;
-            req.url = url.format(rewriteUrlParsed);
+
+            if (isURL(rewritePath)) {
+              const rewriteUrlParsed = new URL(rewritePath);
+              const rewriteOrigin = `${rewriteUrlParsed.protocol}//${rewriteUrlParsed.host}`;
```

Comment:
```suggestion
              const rewriteOrigin = rewriteUrlParsed.origin;
```

Can we use the origin prop?